devtools/gradle/src/main/java/io/quarkus/gradle/AppModelGradleResolver.java

```diff
@@ -140,7 +140,7 @@ public AppModel resolveModel(AppArtifact appArtifact) throws AppModelResolverExc
         final List<AppDependency> fullDeploymentDeps = new ArrayList<>();
         if (!extensionDeps.isEmpty()) {
             final Configuration deploymentConfig = project.getConfigurations()
-                    .detachedConfiguration(extensionDeps.toArray(new Dependency[extensionDeps.size()]));
```

Comment:
Just curious why this line was replaced?

Reply:
I assume because of: https://shipilev.net/blog/2016/arrays-wisdom-ancients/#_conclusion

Reply:
IntellIJ was complaining about it. Apparently this is not an issue anymore in 2020:


>Call to 'toArray()' with pre-sized array argument 'new Dependency[extensionDeps.size()]' 
> Inspection info: There are two styles to convert a collection to an array: either using a pre-sized array (like c.toArray(new String[c.size()])) or using an empty array (like c.toArray(new String[0]).
>In older Java versions using pre-sized array was recommended, as the reflection call which is necessary to create an array of proper size was quite slow. However since late updates of OpenJDK 6 this call was intrinsified, making the performance of the empty array version the same and sometimes even better, compared to the pre-sized version. Also passing pre-sized array is dangerous for a concurrent or synchronized collection as a data race is possible between the size and toArray call which may result in extra nulls at the end of the array, if the collection was concurrently shrunk during the operation.
This inspection allows to follow the uniform style: either using an empty array (which is recommended in modern Java) or using a pre-sized array (which might be faster in older Java versions or non-HotSpot based JVMs).